pkg/resolver/search.go

```diff
@@ -316,45 +316,51 @@ func WhereClauseFilter(ctx context.Context, input *model.SearchInput,
 				klog.V(3).Infof("Property type for [%s] doesn't exist in cache. Refreshing property type cache",
 					filter.Property)
 				propTypeMapNew, err := getPropertyType(ctx, true) // Refresh the property type cache.
-				if err != nil {
-					klog.Errorf("Error creating property type map with err: [%s] ", err)
-					return whereDs, propTypeMap, fmt.Errorf("error [%s] fetching data type for property: [%s]", err, filter.Property)
-				}
-
 				propTypeMap = propTypeMapNew
-
 				dataType, dataTypeInMap = propTypeMap[filter.Property]
 				klog.Infof("For filter prop: %s, datatype is :%s dataTypeInMap: %t\n", filter.Property,
 					dataType, dataTypeInMap)
+				if err != nil || !dataTypeInMap {
+					klog.Errorf("Error creating property type map with err: [%s] or datatype for  [%s] not found in map",
+						err, filter.Property)
+					return whereDs, propTypeMap, fmt.Errorf("error [%s] fetching data type for property: [%s]",
+						err, filter.Property)
+				}
 			}
 
-			if len(propTypeMap) > 0 && dataTypeInMap {
-				klog.V(5).Infof("For filter prop: %s, datatype is :%s\n", filter.Property, dataType)
-
-				// if property matches then call decode function:
-				values, err = decodePropertyTypes(values, dataType)
-				if err != nil {
-					return whereDs, propTypeMap, err
-				}
-				// Check if value is a number or date and get the cleaned up value
-				opDateValueMap = getOperatorAndNumDateFilter(filter.Property, values, dataType)
-			} else {
-				klog.Error("Error with property type list is empty.")
-				values = decodePropertyTypesNoPropMap(values, filter)
-				// Check if value is a number or date and get the cleaned up value
-				opDateValueMap = getOperatorAndNumDateFilter(filter.Property, values, nil)
+			klog.V(5).Infof("For filter prop: %s, datatype is :%s\n", filter.Property, dataType)
 
+			// if property matches then call decode function:
+			values, err = decodePropertyTypes(values, dataType)
+			if err != nil {
+				return whereDs, propTypeMap, err
+			}
+			simpleMatch := true
+			if dataType == "object" || dataType == "array" {
+				opValueMap = extractOperator(values, "@>", opValueMap)
+				simpleMatch = false
 			}
+			// Check if value is a number or date and get the cleaned up value
+			if compareValues(values, []string{"hour", "day", "week", "month", "year"}) {
+				opValueMap = getOperatorIfDateFilter(filter.Property, values, opValueMap)
+				simpleMatch = false
+			}
+			if compareValues(values, []string{"*"}) {
+				opValueMap = getPartialMatchFilter(filter.Property, values, dataType, opValueMap)
+				simpleMatch = false
+			}
+			if simpleMatch {
+				opValueMap = extractOperator(values, "", opValueMap)
+			}
```

Comment:
1. Can we use `if/else if` to make it clear that only 1 should match?
2. Maybe this block could be moved to a function.

Reply:
Done